src/pages/index.js

```diff
@@ -3,104 +3,248 @@ import FormValidator from "../components/FormValidator.js";
     });
-    popupWithEditProfileForm.close();
-    editFormValidator.disableSubmitButton();
 }
 
 function handleAddCardFormSubmit(inputValue) {
-    const cardData = {
-        name: inputValue.title,
-        link: inputValue.url,
-    };
-    renderCard(cardData);
-    popupWithAddCardForm.close();
-    addCardForm.reset();
-    addFormValidator.disableSubmitButton();
+  const cardData = {
+    name: inputValue.title,
+    link: inputValue.url,
+  };
+
+  popupWithAddCardForm.setLoadingText(true);
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      renderCard(newCard);
+    })
+    .finally(() => {
+      popupWithAddCardForm.setLoadingText(false);
+    });
+  popupWithAddCardForm.close();
+  addCardForm.reset();
+  addFormValidator.disableSubmitButton();
+}
 
+function openConfirmPopup(card) {
+  popupWithDeleteButton.setSubmitFunction(() => {
+    // this should be basically the handleDeleteCardConfirm logic
+    api
+      .deleteCardData(card._cardId)
+      .then(() => {
+        // use handleDeleteCard which removes the card from the DOM
+        card.handleDeleteCard(); //
+      })
+      .catch((err) => {
+        console.error("Failed to delete card: ", err);
+      })
+      .finally(() => {
+        popupWithDeleteButton.close();
+      });
+  });
+  // this is the part of the handleDeleteClick
+  popupWithDeleteButton.open();
 }
 
 addCardButton.addEventListener("click", () => {
-    popupWithAddCardForm.open();
+  popupWithAddCardForm.open();
 });
 
 profileEditButton.addEventListener("click", () => {
-    const currentUserInfo = userInfo.getUserInfo();
-    profileTitleInput.value = currentUserInfo.name
-    profileDescriptionInput.value = currentUserInfo.description
-    popupWithEditProfileForm.open();
+  const currentUserInfo = userInfo.getUserInfo();
+  profileTitleInput.value = currentUserInfo.name;
+  profileDescriptionInput.value = currentUserInfo.about;
+  popupWithEditProfileForm.open();
+});
+avatarUpdateButton.addEventListener("click", () => {
+  const currentUserInfo = userInfo.getUserInfo();
+  avatarUrlInput.value = currentUserInfo.avatar;
+  popupWithEditAvatarForm.open();
 });
 
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "196bd4a5-0b68-4944-a5da-dae91687bd99",
+    "Content-Type": "application/json",
+  },
+});
 
-initialCards.forEach((cardData) => renderCard(cardData));
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData); // Display user info in UI
+  })
+  .catch((err) => console.error("Failed to load user info:", err));
+
+//card renderer api
+let cardArray = [];
+
+let data = api.getCardData();
+data.then((data) => {
+  cardArray = data;
+  console.log(cardArray);
+  cardArray.forEach((cardData) => renderCard(cardData));
+});
+
+const section = new Section(
+  {
+    items: cardArray,
+    renderer: renderCard,
+  },
+  ".cards__list"
```

Comment:
`let cardArray = [];` is not needed because you don't call the `renderItems` method for rendering the array of cards


You'll have only this here:

```js
//card renderer api

api
  .getCardData()
  .then((data) => {
    data.forEach((cardData) => renderCard(cardData));
  })
  .catch(console.error);  // don't forget about catching errors

const section = new Section(
  {
    renderer: renderCard,
  },
  ".cards__list"
);

```